lib/pc_patch.c

```diff
@@ -724,3 +724,83 @@ pc_patch_set_schema(PCPATCH *patch, const PCSCHEMA *new_schema, double def)
 
 	return (PCPATCH*) paout;
 }
+
+
+/**
+* Read all the points from "patch", and transform them based on "new_schema".
+* Return a new patch with the transformed points.
+*/
+PCPATCH*
+pc_patch_transform(const PCPATCH *patch, const PCSCHEMA *new_schema, double def)
+{
+	PCDIMENSION** new_dimensions = new_schema->dims;
+	PCDIMENSION* old_dimensions[new_schema->ndims];
+	const PCSCHEMA *old_schema = patch->schema;
+	PCPATCH_UNCOMPRESSED *paout;
+	PCPOINT opt, npt;
+	PCPATCH *pain;
+	size_t i, j;
+
+	if ( old_schema->srid != new_schema->srid )
+	{
+		pcwarn("old and new schemas have different srids, and data "
+			   "reprojection is not yet supported");
+		return NULL;
+	}
+
+	for ( j = 0; j < new_schema->ndims; j++ )
+	{
+		PCDIMENSION *ndim = new_dimensions[j];
+		PCDIMENSION *odim = pc_schema_get_dimension_by_name(
+				old_schema, ndim->name);
+		old_dimensions[j] = odim;
+	}
+
```

Comment:
> I tend to favor to leave like this, merge it and open an issue on the topic for further discussion on whether it should rather fail, continue with the current implementation, or continue with an optimized implementation.

Or instead of an issue that will stay there for ever I can add a comment in the code. It may also stay there for a long time, but I think it's a better place for this sort of things.